File: Programmers/src/JS/같은숫자는싫어.js
function solution(arr)
{
    var answer = [];
    let num = 10;

    for(let i=0; i<arr.length; i++){
        if(arr[i] !== num)
            answer.push(arr[i]);
        num = arr[i];
    }
    
    return answer;
}

File: Programmers/src/Java/Lv1/같은 숫자는 싫어.java
import java.util.*;

public class Solution {
    public int[] solution(int []arr) {
        ArrayList<Integer> list = new ArrayList<>();
        
        int num = -1;
        for(int i=0; i<arr.length; i++) {
            if(arr[i] != num) {
                list.add(arr[i]);
                num = arr[i];
            }
        }
        
        int[] answer = new int[list.size()];
        for(int i=0; i<answer.length; i++)
            answer[i] = list.get(i);
        
        return answer;
    }
}